archive/100-theorems-list/70_perfect_numbers.lean

```diff
@@ -0,0 +1,671 @@
+import data.nat.totient
+import number_theory.quadratic_reciprocity
+
+open_locale classical
+open_locale big_operators
+
+section definitions
+
+variable (n : ℕ)
+
+def divisors : finset ℕ :=
+finset.filter (λ x : ℕ, x ∣ n) (finset.Ico 1 (n + 1))
+
+def proper_divisors : finset ℕ :=
+finset.filter (λ x : ℕ, x ∣ n) (finset.Ico 1 n)
+
+lemma not_proper_self :
+¬ n ∈ proper_divisors n :=
+begin 
+  rw proper_divisors,
+  simp,
+end
```

Comment:
probably `by simp [proper_divisors]` will suffice here?

Reply:
It does.

Reply:
After you commit the change, make sure to mark this comment thread as resolved.